### PR TITLE
NEXUS-42431-Alpine

### PR DESCRIPTION
**Pull request by mburkert3**
This PR:
- Adds the Alpine base image Dockerfile
- Creates a temp Jenkins job to build image and push the result to Sonatype's internal repo

[NEXUS-42431](https://sonatype.atlassian.net/browse/NEXUS-42431)

[NEXUS-42431]: https://sonatype.atlassian.net/browse/NEXUS-42431?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile.alpine [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ARG NEXUS_DOWNLOAD_URL=https://download.sonatype.com/nexus/3/nexus-${NEXUS_VERSION}-unix.tar.gz
+ARG NEXUS_DOWNLOAD_SHA256_HASH=4161a1bb43d92ad8ca6185fa0da2c0f02dfd62280c5b6e4ac4419df4aecaf55f
+
+# configure nexus runtime
+ENV SONATYPE_DIR=/opt/sonatype
+ENV NEXUS_HOME=${SONATYPE_DIR}/nexus \
+    NEXUS_DATA=/nexus-data \
+    NEXUS_CONTEXT='' \
+    SONATYPE_WORK=${SONATYPE_DIR}/sonatype-work \
+    DOCKER_TYPE='rh-docker'
```
Comment (nblair) [inline Dockerfile.alpine:49]: I think the value of this argument should differ, use 'alpine' instead.

Comment (mburkert3) [reply Dockerfile.alpine]: ✅

**Review by Alejandro-Tovar — APPROVED** [review]

File: Jenkinsfile.alpine [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      def hash = OsTools.runSafe(this, "docker build --quiet --label base-image-ref='${baseImageReferenceStr}' --no-cache --tag ${imageName} . -f ${dockerfilePath}")
+      imageId = hash.split(':')[1]
+    }
+    if (params.scan_for_policy_violations) {
+      stage('Evaluate Policies') {
+        runEvaluation({ stage ->
+          nexusPolicyEvaluation(
+              iqStage: stage,
+              iqApplication: 'docker-nexus3',
+              iqScanPatterns: [[scanPattern: "container:${imageName}"]],
```
Comment (jlhuerfanor) [inline Jenkinsfile.alpine:68]: We have a new application in IQ for alpine-based image: https://sonatype.sonatype.app/platform/assets/index.html#/management/view/application/docker-nexus3-alpine

**Review by jlhuerfanor — CHANGES_REQUESTED** [review]

File: Jenkinsfile.alpine [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      def baseImageReferenceStr = baseImageReference.getReference()
+      def hash = OsTools.runSafe(this, "docker build --quiet --label base-image-ref='${baseImageReferenceStr}' --no-cache --tag ${imageName} . -f ${dockerfilePath}")
+      imageId = hash.split(':')[1]
+    }
+    if (params.scan_for_policy_violations) {
+      stage('Evaluate Policies') {
+        runEvaluation({ stage ->
+          nexusPolicyEvaluation(
+              iqStage: stage,
+              iqApplication: 'docker-nexus3',
```
Comment (jlhuerfanor) [inline Jenkinsfile.alpine:67]: We have a new IQ application in lifecycle for alpine-images: https://sonatype.sonatype.app/platform/assets/index.html#/management/view/application/docker-nexus3-alpine

```suggestion
              iqApplication: 'docker-nexus3-alpine',
```

**Review by jlhuerfanor — APPROVED** [review]

**Pull request merged**